plugins/inventory/vmware_vm_inventory.py

```diff
@@ -624,6 +645,18 @@ def _get_hostname(self, properties, hostnames):
             if hostname:
                 return to_text(hostname)
 
+    def _can_add_host(self, host_filters, host_properties, host, strict=False):
+        for host_filter in host_filters:
+            try:
+                can_add_host = self._compose(host_filter, host_properties)
+            except Exception as e:  # pylint: disable=broad-except
```

Comment:
The fact that the exceptions hard hidden by default (`strict=False`) make me sad.

Reply:
This pattern is all over the base constructable class

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/__init__.py

what this does should really be a common pattern from Ansible core. Server-side filtering should be preferred, but client-side filtering shouldn't need to be re-implemented by every collection using this. I don't think there should be any issue with the blanket exception here, because that's now Ansible core does it.

Reply:
I agree with @AlanCoding. When the user specifies `strict=False`, the user is OK with exceptions ignored.